server/src/main/java/org/opensearch/extensions/ExtensionDependency.java

```diff
@@ -54,6 +60,39 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVersion(version);
     }
 
+    public static ExtensionDependency parse(XContentParser parser) throws IOException {
+        String uniqueId = null;
+        Version version = null;
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            switch (fieldName) {
+                case UNIQUE_ID:
+                    uniqueId = parser.text();
+                    break;
+                case VERSION:
+                    try {
+                        version = Version.fromString(parser.text());
```

Comment:
Why only try/catch here? 

Perhaps something like above^

Reply:
Because `fromString` throws an exception [here](https://github.com/opensearch-project/OpenSearch/blob/main/libs/core/src/main/java/org/opensearch/Version.java#L169) which needs to be catch.

Rest of the exceptions including uniqueId is handled below.